### PR TITLE
Use dollar street dataset

### LINKED ISSUE #445: CIFAR10 not having a license
As just discussed [here](https://github.com/carpentries-lab/reviews/issues/25#issuecomment-1965113607) it might be worth considering to replace CIFAR-10 for some other dataset.

### PR DESCRIPTION
**Pull request by svenvanderburg**
Fixes #445

### CODE REVIEW

File: episodes/4-advanced-layer-types.Rmd [hunk 1]
```diff
@@ -27,45 +27,35 @@ So far, we have seen one type of layer, namely the **fully connected**, or **den
...
-:::
-
-## CIFAR-10
-
-The CIFAR-10 dataset consists of images of 10 different classes: airplanes, cars, birds, cats, deer, dogs, frogs, horses, ships, and trucks.
-It is widely used as a benchmark dataset for image classification. The low resolution of the images in the dataset allows for quick loading and testing models.
-
-For more information about this dataset and how it was collected you can check out
-[Learning Multiple Layers of Features from Tiny Images by  Alex Krizhevsky, 2009](https://www.cs.toronto.edu/~kriz/learning-features-2009-TR.pdf).
+The MLCommons Dollar Street Dataset is a collection of images of everyday household items from homes around the world that visually captures socioeconomic diversity of traditionally underrepresented populations. We use a subset of the original dataset that can be used for multiclass classification with 10 categories. Let's load the data:
```
Comment (tobyhodges) [inline episodes/4-advanced-layer-types.Rmd:30]:
```suggestion
The [MLCommons Dollar Street Dataset](https://www.kaggle.com/datasets/mlcommons/the-dollar-street-dataset) is a collection of images of everyday household items from homes around the world that visually captures socioeconomic diversity of traditionally underrepresented populations. We use [a subset of the original dataset](https://zenodo.org/records/10837090) that can be used for multiclass classification with 10 categories. Let's load the data:
```

File: episodes/4-advanced-layer-types.Rmd [hunk 2]
```diff
@@ -27,45 +27,35 @@ So far, we have seen one type of layer, namely the **fully connected**, or **den
...
+DATA_FOLDER = pathlib.Path('data/dataset_dollarstreet/') # change to location where you stored the data
+train_images = np.load(DATA_FOLDER / 'train_images.npy')
+val_images = np.load(DATA_FOLDER / 'test_images.npy')
+train_labels = np.load(DATA_FOLDER / 'train_labels.npy')
+val_labels = np.load(DATA_FOLDER / 'test_labels.npy')
+```
+
+## Dollar Street 10 dataset
+
+The dollar street 10 dataset consists of images of 10 different classes, this is the mapping of the categories:
```
Comment (tobyhodges) [inline episodes/4-advanced-layer-types.Rmd:43]:
```suggestion
The dollar street 10 dataset consists of images of 10 different classes, this is the mapping of the categories:

```

This should make the table render correctly. It was showing up unformatted on my local build.

File: episodes/4-advanced-layer-types.Rmd [hunk 3]
```diff
@@ -27,45 +27,35 @@ So far, we have seen one type of layer, namely the **fully connected**, or **den
...
+| running shoe    | 3     |
+| soap dispenser  | 4     |
+| street sign     | 5     |
+| table lamp      | 6     |
+| tile roof       | 7     |
+| toilet seat     | 8     |
+| washing machine | 9     |
+
+
+![Sample images from the dollar street 10 data-set. Each image is labelled with a category, for example: 'street sign' or 'soap dispenser'](fig/04_dollar_street_10){alt="A 5 by 5 grid of 25 sample images from the dollar street 10 data-set. Each image is labelled with a category, for example: 'street sign' or 'soap dispenser'."}
```
Comment (tobyhodges) [inline episodes/4-advanced-layer-types.Rmd:58]:
```suggestion
![Sample images from the dollar street 10 data-set. Each image is labelled with a category, for example: 'street sign' or 'soap dispenser'](fig/04_dollar_street_10){alt="A 5 by 5 grid of 25 sample images from the dollar street 10 data-set. Each image is labelled with a category, for example: 'street sign' or 'soap dispenser'."}

```

An empty line to make the section heading display correctly

File: episodes/4-advanced-layer-types.Rmd [hunk 4]
```diff
@@ -27,45 +27,35 @@ So far, we have seen one type of layer, namely the **fully connected**, or **den
...
 ## 2. Identify inputs and outputs
 
 ## Explore the data
```
Comment (tobyhodges) [inline episodes/4-advanced-layer-types.Rmd:61]:
```suggestion
### Explore the data
```

I think this and some below should be third-level headings, to present the heading hierarchy correctly.

File: episodes/4-advanced-layer-types.Rmd [hunk 5]
```diff
@@ -75,24 +65,24 @@ Let's do a quick exploration of the dimensions of the data:
...
-The remainder of the shape, namely `32, 32, 3)`, denotes
+The first value, `878`, is the number of training images in the dataset.
+The remainder of the shape, namely `128, 128, 3)`, denotes
 the dimension of one image. The last value 3 is typical for color images,
 and stands for the three color channels **R**ed, **G**reen, **B**lue.
 
 ::: challenge
 
-## Number of features CIFAR-10
+## Number of features dollar-street-10
```
Comment (tobyhodges) [inline episodes/4-advanced-layer-types.Rmd:78]:
```suggestion
### Number of features dollar-street-10
```

File: episodes/4-advanced-layer-types.Rmd [hunk 6]
```diff
@@ -471,35 +454,35 @@ dense_model.summary()
...
 compile_model(dense_model)
-history = dense_model.fit(train_images, train_labels, epochs=30,
+history = dense_model.fit(train_images, train_labels, epochs=20,
                     validation_data=(val_images, val_labels))
 plot_history(history, ['accuracy', 'val_accuracy'])
 ```
 ![](fig/04_dense_model_training_history.png){alt="Plot of training accuracy and validation accuracy vs epochs for a model with only dense layers"}
 
-As you can see the validation accuracy only reaches about 35%, whereas the CNN reached about 55% accuracy.
+As you can see the validation accuracy only reaches about 12,5%, whereas the CNN reached about 25% accuracy.
```
Comment (tobyhodges) [inline episodes/4-advanced-layer-types.Rmd:485]:
```suggestion
As you can see the validation accuracy only reaches about 12.5%, whereas the CNN reached about 25% accuracy.
```

**Review by tobyhodges — COMMENTED** [review]
Suggesting a few minor changes as I read through what you have changed already. It looks very promising @svenvanderburg. Would you like me to jump in and adjust the remaining code/output blocks (e.g. changing model names) and inserting the remaining plots that you have not already updated?

File: episodes/4-advanced-layer-types.Rmd [hunk 7]
```diff
@@ -27,45 +27,35 @@ So far, we have seen one type of layer, namely the **fully connected**, or **den
...
-
-
+import pathlib
+DATA_FOLDER = pathlib.Path('data/dataset_dollarstreet/') # change to location where you stored the data
+train_images = np.load(DATA_FOLDER / 'train_images.npy')
+val_images = np.load(DATA_FOLDER / 'test_images.npy')
+train_labels = np.load(DATA_FOLDER / 'train_labels.npy')
+val_labels = np.load(DATA_FOLDER / 'test_labels.npy')
+```
+
```
Comment (tobyhodges) [inline episodes/4-advanced-layer-types.Rmd:40]:
```suggestion

:::::::::::::::::::::::::::::::::::::::::: callout

### A Note About Data Provenance
In an earlier version, this part of the lesson used a different example dataset.
During [peer review](https://github.com/carpentries-lab/reviews/issues/25#issuecomment-1953271802), the decision was made to replace that dataset due to the way it had been compiled using images "scraped" from the internet without permission from or credit to the original creators of those images. Unfortunately, uncredited use of images is a common problem among datasets used to benchmark models for image classification.
The Dollar Street Dataset was chosen for use in the lesson as it contains only images [created by the Gapminder project](https://www.gapminder.org/dollar-street/about?) expressly for the purposes of use in the dataset.
The original Dollar Street Dataset is very large -- more than 100 GB -- with the potential to grow even bigger, so we created a subset for use in this lesson.

::::::::::::::::::::::::::::::::::::::::::::::::::


```

Suggesting this to address https://github.com/carpentries-incubator/deep-learning-intro/issues/445#issuecomment-1988375148

Comment (svenvanderburg) [reply episodes/4-advanced-layer-types.Rmd:40]: Love it!!!

**Comment by svenvanderburg** [comment]
Thanks for your great suggestions @tobyhodges! I really like the part about provenance of the dataset👍 Yes you can pick this up if you want, but keep in mind that adding the images of all the graphs is quite a hassle :) If you don't pick it up I will pick it up in the week of 15th of April. But very much appreciated if you do it! 👍

File: episodes/4-advanced-layer-types.Rmd [hunk 8]
```diff
@@ -27,44 +27,47 @@ So far, we have seen one type of layer, namely the **fully connected**, or **den
...
+| running shoe    | 3     |
+| soap dispenser  | 4     |
+| street sign     | 5     |
+| table lamp      | 6     |
+| tile roof       | 7     |
+| toilet seat     | 8     |
+| washing machine | 9     |
+
+
+![Sample images from the dollar street 10 data-set. Each image is labelled with a category, for example: 'street sign' or 'soap dispenser'](fig/04_dollar_street_10){alt="A 5 by 5 grid of 25 sample images from the dollar street 10 data-set. Each image is labelled with a category, for example: 'street sign' or 'soap dispenser'."}
```
Comment (tobyhodges) [inline episodes/4-advanced-layer-types.Rmd:70]:
```suggestion
![Sample images from the dollar street 10 data-set. Each image is labelled with a category, for example: 'street sign' or 'soap dispenser'](fig/04_dollar_street_10.png){alt="A 5 by 5 grid of 25 sample images from the dollar street 10 data-set. Each image is labelled with a category, for example: 'street sign' or 'soap dispenser'."}
```

File: episodes/4-advanced-layer-types.Rmd [hunk 9]
```diff
@@ -471,35 +467,35 @@ dense_model.summary()
...
 Non-trainable params: 0 (0.00 Byte)
 _________________________________________________________________
 ```
-As you can see this model has 1.5x more parameters than our simple CNN, let's train and evaluate it!
+As you can see this model has more parameters than our simple CNN, let's train and evaluate it!
 
 ```python
 compile_model(dense_model)
-history = dense_model.fit(train_images, train_labels, epochs=30,
+history = dense_model.fit(train_images, train_labels, epochs=20,
```
Comment (tobyhodges) [inline episodes/4-advanced-layer-types.Rmd:492]: A couple of observations when I got to this point:

1. This model trained _very_ quickly
2. The training history plot looks weird...

04_dense_model_training_history

**Comment by tobyhodges** [comment]
@svenvanderburg I have pushed a few more changes. Please take a look if and when you have time.

**Comment by svenvanderburg** [comment]
@tobyhodges great changes!

While doing some minor changes I realised that the training really takes too long for teaching. I ended up using smaller images (64 x 64 instead of 128 x 128). This speeds up training to a workable duration and reduces overfitting a bit.

You can checkout the related changes here: https://github.com/carpentries-incubator/deep-learning-intro/pull/448/files/40b7c7e28174532002751aaa2f6fe4878a2020a5..8a3e077d56c516eb8b7abdc282485b948334dc17

I also updated the setup instructions to instruct learners to download the dataset.

I feel like this is ready to go now, can you have a final look?

**Review by tobyhodges — APPROVED** [review]
Suggesting one very minor change but otherwise this is great. Thanks for all your work on it @svenvanderburg.

I think the episode is missing one or two concluding sentences at the very end, summarising what we achieved (and did not achieve) with the models trained here, lessons learned (e.g. about training data and the difficulty of image classification tasks) and perhaps suggesting some next steps the learner could take if they wanted to improve their results. But that can be added in a separate PR.
